core/src/main/java/com/amplifyframework/core/AmplifyConfiguration.java

```diff
@@ -169,13 +169,14 @@ private static JSONObject readInputJson(Context context, int resourceId) throws
     public CategoryConfiguration forCategoryType(@NonNull CategoryType categoryType) throws AmplifyException {
         final CategoryConfiguration categoryConfiguration =
             categoryConfigurations.get(categoryType.getConfigurationKey());
-        if (categoryConfiguration == null) {
-            throw new AmplifyException(
-                "Unknown/bad category type: " + categoryType,
-                "Be sure to use one of the supported Categories in your current version of Amplify"
-            );
-        }
-        return categoryConfiguration;
+        return categoryConfiguration != null ? categoryConfiguration :
```

Comment:
Will this return a dummy `CategoryConfiguration` for any key I pass? For example, if I pass a key for category `pretend`, I'll get back a Null-y object for it?

Reply:
You can only pass one of the members of the `CategoryType` enum. It will return an empty CategoryConfiguration instance of the type you requested if one was no loaded via the config file. 

This is also related to handling plugins that get added by the developer without corresponding entries in the config file, which seems to be the case with DataStore for the most part since the CLI does not add that to the config file. Perhaps this is something we can handle in a separate PR?